resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -583,6 +597,47 @@ private object YarnClasspathTest extends Logging {
 
 }
 
+private object YarnAddJarTest extends Logging {
```

Comment:
There's a lot of duplicated code between all of these test apps for argument parsing / validation and writing out a `result` file. I think we should consider doing a little refactoring to allow them to inherit from a trait which handles some of this shared logic, or to leverage common utility methods.

But I have no complaints if you want to maintain the status quo in this patch instead of trying to tackle that as well. Maybe a good follow-on PR.

Reply:
Given the amount of discussion around the primary change itself, I think its best to keep this as a followup. I have added a note in SPARK-34472.

Reply:
+1